File: ios/Sources/TrainAlarm/Models/Station.swift
import Foundation

/// A physical station. `id` is provider-specific (e.g. a CRS code for UK
/// rail) - treat it as an opaque identifier, not a fixed format, since a
/// future non-UK provider will use a different scheme.
struct Station: Identifiable, Equatable, Hashable {
    let id: String
    let name: String
    let latitude: Double
    let longitude: Double
}

File: android/app/src/main/java/com/trainalarm/app/model/Station.kt
package com.trainalarm.app.model

/**
 * A physical station. [id] is provider-specific (e.g. a CRS code for UK
 * rail) - callers should treat it as an opaque identifier, not assume a
 * format, since a future non-UK provider will use a different scheme.
 */
data class Station(
    val id: String,
    val name: String,
    val latitude: Double,
    val longitude: Double
)
